kuksa_databroker/databroker/src/glob.rs

```diff
@@ -51,3 +52,70 @@ pub fn to_regex(glob: &str) -> Result<Regex, Error> {
     let re = to_regex_string(glob);
     Regex::new(&re).map_err(|_err| Error::RegexError)
 }
+
+pub fn to_regex_string_new(glob: &str) -> String {
+    // Construct regular expression
+
+    // Make sure we match the whole thing
+    // Start from the beginning
+    let mut re = String::from("^");
+
+    // Replace all "standalone" wildcards with ".*"
+    re.push_str(
+        &glob
+            .split('.')
+            .map(|part| match part {
+                "*" => "[^.]+",
+                "**" => ".*",
+                other => other,
+            })
+            .collect::<Vec<&str>>()
+            .join(r"\."),
+    );
+    // And finally, make sure we match until EOL
+    re.push('$');
+
+    re
+}
+
+pub fn to_regex_new(glob: &str) -> Result<Regex, Error> {
+    let re = to_regex_string_new(glob);
+    Regex::new(&re).map_err(|_err| Error::RegexError)
+}
+
+pub fn matches_path_pattern(input: &str) -> bool {
+    let pattern = r"^(?:\w+(?:\.\w+)*)(?:\.\*+)?$";
+    let regex = Regex::new(pattern).unwrap();
+
+    let pattern2 = r"^(\*{2}|\*\.)?[^.]+(\.[^.]+)*\.(\*{2}|\*|[A-Za-z]+)(\.[^.]+)*\.[^.]+$";
+    let regex2 = Regex::new(pattern2).unwrap();
+
+    regex.is_match(input) || regex2.is_match(input)
```

Comment:
These regex:es do not seem to correctly identify valid patterns.

You can have a look at the regex in `src/jwt/scope.rs`, as that covers the same functionality (i.e. the `path` group matches valid patterns).